schemas/2016-06-27-preview/Microsoft.ContainerRegistry.json

```diff
@@ -34,60 +39,82 @@
               "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
             }
           ],
-          "description": "Gets or sets a list of key value pairs that describe the resource. These tags can be used in viewing and grouping this resource (across resource groups). A maximum of 15 tags can be provided for a resource. Each tag must have a key no greater than 128 characters and value no greater than 256 characters."
+          "description": "The tags of the resource."
         },
         "properties": {
           "oneOf": [
             {
-              "$ref": "#/definitions/ContainerRegistryProperties"
+              "$ref": "#/definitions/RegistryProperties"
             },
             {
               "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
             }
-          ]
+          ],
+          "description": "The properties of the container registry."
         }
       },
       "required": [
-        "apiVersion",
+        "name",
         "type",
+        "apiVersion",
         "location",
         "properties"
-      ]
+      ],
+      "description": "Microsoft.ContainerRegistry/registries"
     }
   },
   "definitions": {
-    "ContainerRegistryProperties": {
+    "RegistryProperties": {
       "type": "object",
       "properties": {
-        "storageAccount": {
-          "type": "object",
-          "properties": {
-            "name": {
-              "type": "string"
+        "adminUserEnabled": {
+          "oneOf": [
+            {
+              "type": "boolean",
+              "enum": [
+                "false"
```

Comment:
>"false" [](start = 16, length = 7)

Is this correct? This is a string value, while the type says "boolean".

Reply:
same in the other api versions. 

---
In reply to: [148637443](https://github.com/Azure/azure-resource-manager-schemas/pull/370#discussion_r148637443) [](ancestors = 148637443)

Reply:
This does seem odd, but is this AutoRest behavior? I did see the same issue in other schemas such as https://github.com/Azure/azure-resource-manager-schemas/blob/0f742766eb22be8d2807283e8055221d373ebf79/schemas/2017-06-01/Microsoft.Storage.json#L467
What is the workaround for this issue?

Reply:
In fact, the issue is not only that `"false"` is a string, but it should also be an enum with both `true` and `false` values.

Reply:
Right. can you check your swagger file and see if it has the correct definition?

Reply:
I suppose so.
https://github.com/Azure/azure-rest-api-specs/blob/current/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/2017-10-01/containerregistry.json#L1206
It seems that AutoRest is not treating `boolean` with default value properly.

Reply:
@Tiano2017 Any update on this? Since this seems to be an AutoRest issue and it has been spread in published schema for many other services, can we have this merged and have a general cleanup later on? If not possible, we will have to request an exception of delaying the release of ACR schema as we had committed to have them a few weeks ago. Thanks. /cc @sajayantony

Reply:
works for me. I'll merge this PR.